### PR TITLE
[FIXED JENKINS-26234]  Adding crumb exclusion

### PR DESCRIPTION
**Pull request by mcgin**
Implementation copied from the mercurial plugin - https://github.com/jenkinsci/mercurial-plugin/pull/71

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/BitbucketHookReceiver.java [hunk 1]
```diff
@@ -60,4 +66,14 @@ public void doIndex(StaplerRequest req) throws IOException {
 
     private static final Logger LOGGER = Logger.getLogger(BitbucketHookReceiver.class.getName());
 
+    @Override
+    public boolean process(HttpServletRequest req, HttpServletResponse resp, FilterChain chain)
+            throws IOException, ServletException {
+        String pathInfo = req.getPathInfo();
+        if (pathInfo != null && pathInfo.startsWith("/"+BITBUCKET_HOOK_URL)) {
```
Comment (fbelzunc) [inline src/main/java/com/cloudbees/jenkins/plugins/BitbucketHookReceiver.java]: I prefer you create:

```
private static final String EXCLUSION_PATH = "/" + BitbucketHookReceiver. BITBUCKET_HOOK_URL;
```

And then use:

```
 if (pathInfo != null && (pathInfo.equals(EXCLUSION_PATH) || pathInfo.equals(EXCLUSION_PATH + "/")))
```

**Comment by fbelzunc** [comment]
Please, use a new class for this called `BitbucketCrumbExclusion`

```
@Extension
public class BitbucketCrumbExclusion extends CrumbExclusion {
```

**Comment by fbelzunc** [comment]
:ant Main important thing is that you create a new class in order to make the improvement more clear. 

If possible, please, change the title of the pull request to something like this:

```
[FIXED JENKINS-26234]  Adding crumb exclusion
```

**Comment by mcgin** [comment]
Have moved the exclusion to it's own class as requested along with the other cleanup.

**Comment by fbelzunc** [comment]
It is now merged.

Many thanks!

**Pull request merged**